### ISSUE TITLE
Change Permissions Settings ⚠️

### ISSUE THREAD

**Issue by hellyeahmfs** [marchellodev/sharik#100]
The app requires permission to manage 'All Files'. Instead of this the app should allow user to select file from file picker. I don't know exactly which term to use 😂 but i think you can get my point. Managing all files is not good for security imo.

**Comment by marchellodev** [comment]
100th issue!!! :fireworks: :champagne: 

Thanks! That's actually a very good idea, I don't actually think we need to be able to access all files, since they are cached in app's storage anyways. I'm going to research this a bit more though.

Thanks again :)

**Comment by hellyeahmfs** [comment]
Yeah Sharik can do that like this app - [Just Player](https://github.com/moneytoo/Player) and Sharik can save received files in a particular directory that Sharik has only access to like this app - [Barinsta](https://codeberg.org/avalos/barinsta). I hope it can help you in any way 😊

**Comment by marchellodev** [comment]
It seems like this permission is necessary for the app to be able to share files.
[See this](https://stackoverflow.com/questions/23692675/permission-denial-when-starting-intent-using-action-get-content-with-set-type)

**Comment by hellyeahmfs** [comment]
[This app](https://github.com/fm-sys/snapdrop-android) can do that.

**Comment by marchellodev** [comment]
@hellyeahmfs I have only one android device near me, are there any similar apps that do that?

**Comment by marchellodev** [comment]
Also, I have just noticed that the minimal verison of android that Snapdrop supports is 5 (` minSdkVersion 21`). However, we target android 4 as well. So, as far as I understand, this permission is necessary for the app to function on android 4. Please see the link I provided earlier.

**Comment by hellyeahmfs** [comment]
Okay. But i think supporting Android 4 is not very much important because most of the people don't use that (and imo people don't use Android 5 too 😂)

This feature is going to increase security and privacy very much and thats a very good sign for an app.

If you want to still support Android 4 and don't want to implemant it for now thats your choice :-)

**Comment by hellyeahmfs** [comment]
[Share via HTTP](https://github.com/marcosdiez/shareviahttp) app works without these permissions and it works just like Sharik.

**Comment by marchellodev** [comment]
> [Share via HTTP](https://github.com/marcosdiez/shareviahttp) app works just like Sharik without these permissions.

Yeah, it also seems to support android 4 ('minSdkVersion 14'), I'll take a look at it a bit later.

Thanks!
